GoRogue/MapGeneration/PolygonArea.cs

```diff
@@ -130,7 +184,115 @@ private void SetInnerPoints()
         }
+        #endregion
+
+        #region static creation methods
+        /// <summary>
+        /// Creates a new Region from a GoRogue.Rectangle.
+        /// </summary>
+        /// <param name="r">The rectangle</param>
+        /// <param name="algorithm">Line-drawing algorithm to use for finding boundaries.</param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static PolygonArea Rectangle(Rectangle r, Lines.Algorithm algorithm = Lines.Algorithm.DDA)
+            => new PolygonArea(algorithm, r.MinExtent, (r.MaxExtentX, r.MinExtentY), r.MaxExtent,
+                (r.MinExtentX, r.MaxExtentY));
+
+        /// <summary>
+         /// Creates a new Region in the shape of a parallelogram, with diagonals going down and right.
+         /// </summary>
+         /// <param name="origin">Origin of the parallelogram.</param>
+         /// <param name="width">Width of the parallelogram.</param>
+         /// <param name="height">Height of the parallelogram.</param>
+        /// <param name="fromTop">Whether the parallelogram extends downward-right or upwards-right from the start</param>
+         /// <param name="algorithm">Line-drawing algorithm to use for finding boundaries.</param>
+         /// <returns>A new region in the shape of a parallelogram</returns>
+         public static PolygonArea Parallelogram(Point origin, int width, int height, bool fromTop = false,
+            Lines.Algorithm algorithm = Lines.Algorithm.DDA)
+         {
+             if (fromTop && Direction.YIncreasesUpward)
+                 height *= -1;
+
+             else if(!fromTop && !Direction.YIncreasesUpward)
+                 height *= -1;
+
+
+             Point p1 = origin;
+             Point p2 = origin + new Point(width, 0);
+             Point p3 = origin + new Point(width * 2, height);
+             Point p4 = origin + new Point(width, height);
+
+             return new PolygonArea(algorithm, p1, p2, p3, p4);
+         }
+
+        /// <summary>
+        /// Creates a polygon whose sides are even-length
+        /// </summary>
+        /// <param name="center">The center point of this polygon</param>
+        /// <param name="numberOfSides">Number of sides and corners on this polygon</param>
+        /// <param name="radius">The desired distance between the center and each corner</param>
+        /// <param name="algorithm">Which line-drawing algoirthm to use</param>
+        /// <returns></returns>
+        public static PolygonArea RegularPolygon(Point center, int numberOfSides, double radius, Lines.Algorithm algorithm = Lines.Algorithm.DDA)
+        {
+            if(numberOfSides < 3)
+                throw new ArgumentException("Polygons must have 3 or more sides to be representable in 2 dimensions");
+
+            if (algorithm == Lines.Algorithm.Bresenham || algorithm == Lines.Algorithm.Orthogonal)
+                throw new ArgumentException("Line Algorithm must produce ordered lines.");
```

Comment:
There was a performance consideration there. I figured that we could throw this exception BEFORE the heavy calculations.

Reply:
Fair enough.  There's a bit of refactoring that could be done to future-proof, but it involves re-adding code similar to something we already had in a previous PR... So let's leave it as-is for now.  This is functionally operational, and if it becomes that much of a bother I can fix it later since one of my suggestions on the last PR is what caused the issue.